services/priceService.ts

```diff
@@ -62,23 +61,121 @@ export async function upsertCurrentPricesForNetworkId (responseData: IResponseDa
   })
 }
 
-export async function syncCurrentPrices (): Promise<boolean> {
+export async function upsertCurrentPricesForNetworkId (responseData: IResponseData, networkId: number): Promise<void> {
+  await upsertPricesForNetworkId(responseData, networkId, 0)
+}
+
+function validateTicker (ticker: string): void {
+  if (!Object.values(TICKERS).includes(ticker)) {
+    throw new Error(RESPONSE_MESSAGES.INVALID_TICKER_400.message)
+  }
+}
+
+function validatePriceAPIUrlAndToken (): void {
   if (appInfo.priceAPIURL === '') {
     throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_URL_400.message)
   }
-  const todayString = moment().format(PRICE_API_DATE_FORMAT)
+  if (appInfo.priceAPIToken === '') {
+    throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_TOKEN_400.message)
+  }
+}
+
+function getPriceByDayURL (day: moment.Moment, ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/pricebydate/${appInfo.priceAPIToken}/${ticker}+${day.format(PRICE_API_DATE_FORMAT)}`
+}
+
+function getAllPricesByTickerURL (ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/dailyprices/${appInfo.priceAPIToken}/${ticker}`
+}
+
+export async function getPriceForDayTicker (day: moment.Moment, ticker: string, attempt: number = 1): Promise<IResponseData | null> {
+  try {
+    const res = await axios.get(getPriceByDayURL(day, ticker), {
+      timeout: PRICE_API_TIMEOUT
+    })
+
+    if (res.data.success !== false) { return res.data } else { return null }
+  } catch (error) {
+    console.error(`Problem getting price of ${ticker} ${day.format('YYYYMMDD')} -> ${error as string} (attempt ${attempt})`)
+
+    if (attempt < PRICE_API_MAX_RETRIES) {
+      return await getPriceForDayTicker(day, ticker, attempt + 1)
+    } else { return null }
+  }
+}
+
+export async function getAllPricesByTicker (ticker: string, attempt: number = 1): Promise<IResponseDataDaily[] | null> {
+  try {
+    const res = await axios.get(getAllPricesByTickerURL(ticker), {
+      timeout: PRICE_API_TIMEOUT
+    })
+
+    if (res.data.success !== false) {
+      const dailyPrices: IResponseDataDaily[] = Object.entries<IResponseData>(res.data).map(e => {
+        return {
+          day: e[0],
+          ...e[1]
+        }
+      })
+      return dailyPrices
+    } else { return null }
+  } catch (error) {
+    console.error(`Problem getting price of ${ticker} -> ${error as string} (attempt ${attempt})`)
+
+    if (attempt < PRICE_API_MAX_RETRIES) {
+      return await getAllPricesByTicker(ticker, attempt + 1)
+    } else { return null }
+  }
+}
+
+export async function syncPastPrices (): Promise<void> {
+  const lastTimestamp = await prisma.price.findFirst({
+    where: {
+      networkId: XEC_NETWORK_ID, // any valid network
```

Comment:
If we don't care about `networkId` and `quoteId`, can we just omit the `where` clause entirely?

Reply:
indeed! done